frappe/automation/doctype/assignment_rule/assignment_rule.py

```diff
@@ -108,13 +119,42 @@ def get_assignments(doc):
 	return frappe.get_all('ToDo', fields = ['name', 'assignment_rule'], filters = dict(
 		reference_type = doc.get('doctype'),
 		reference_name = doc.get('name'),
-		status = 'Open'
+		status = ('!=', 'Cancelled')
 	), limit = 5)
 
-def apply(doc, method):
+@frappe.whitelist()
+def bulk_apply(doctype, docnames):
+	import json
+	docnames = json.loads(docnames)
+
+	background = True #len(docnames) > 5
+	for name in docnames:
+		if background:
+			frappe.enqueue('frappe.automation.doctype.assignment_rule.assignment_rule.apply', doc=None, doctype = doctype, name=name)
+		else:
+			apply(None, doctype = doctype, name=name)
+
+def reopen_closed_assignment(doc):
+	try:
+		todo = frappe.get_list('ToDo', dict(
```

Comment:
```suggestion
		todo = frappe.db.exists('ToDo', dict(
```

Remove `try - except`

Reply:
we need the try except as the todo name we get from get_list is used later in the code